### PR TITLE
Create Target_sum_pairs.java

### PR DESCRIPTION
**Pull request by anahita-singla**
Fixes #

#### Checklist:
- [✓ ] 4 space indentation.
- [✓ ] Coding conventions are followed.
- [✓ ] Input is taken dynamically.
- [✓ ] Sample Input / Output is added at the end of file.
- [✓ ] Logic Documentation (Comments).
- [✓ ] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
- 43
- 

#### Languages Used:
- Java
- 

#### Files Added:
- Target_sum_pairs.java
- 

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

File: 1-D_Array/Java/Target_sum_pairs.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
+//Java program to implement a program which prints all pairs of numbers which sum to target in an array.
+import java.util.*;
+public class Main {
```
Comment (SakshiShreya) [inline 1-D_Array/Java/Target_sum_pairs.java]: There should be a new-line after import statements.

File: 1-D_Array/Java/Target_sum_pairs.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+	public static void main(String args[]) {
```
Comment (SakshiShreya) [inline 1-D_Array/Java/Target_sum_pairs.java]: Use 4 space indentation.

File: 1-D_Array/Java/Target_sum_pairs.java [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+		Scanner scn=new Scanner(System.in);
```
Comment (SakshiShreya) [inline 1-D_Array/Java/Target_sum_pairs.java]: Add space around operators.

File: 1-D_Array/Java/Target_sum_pairs.java [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+		int n=scn.nextInt();
+		int[] a=new int[n];
+		for(int i=0;i<n;i++)
+			a[i]=scn.nextInt();
+		int target=scn.nextInt();
+	    //sort array to apply 2 pointer approach
```
Comment (SakshiShreya) [inline 1-D_Array/Java/Target_sum_pairs.java]: Fix indentation in this line.

File: 1-D_Array/Java/Target_sum_pairs.java [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+			//if the sum is less than target ,increment to find sum with larger number
+			else if(sum<target){
+				start++;
+			}
+			//if the sum is more than target ,decrement to find sum with smaller number
+			else end--;
+		}
+	}
+}
+/* Sample Input:
```
Comment (SakshiShreya) [inline 1-D_Array/Java/Target_sum_pairs.java]: Add a new-line above the same input.

File: 1-D_Array/Java/Target_sum_pairs.java [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+		int n=scn.nextInt();
+		int[] a=new int[n];
+		for(int i=0;i<n;i++)
+			a[i]=scn.nextInt();
+		int target=scn.nextInt();
+	    //sort array to apply 2 pointer approach
+		Arrays.sort(a);
+		//declare 2 pointers at beginning and end of the array
+		int end=n-1,start=0;
+		while(end>start){
```
Comment (SakshiShreya) [inline 1-D_Array/Java/Target_sum_pairs.java]: Add a space between `)` and `{`

**Comment by MastersAbh** [comment]
Hi @anahita-singla Please mention issue number this PR is related to and squash your commits. Code looks fine

**Comment by somya-kapoor** [comment]
Are you working on it?

**Pull request closed without merging**